### PR TITLE
feat(provider): Add the deploy tool

### PR DESCRIPTION
**Pull request by hydrogen18**
I made the following changes from the work that @anilCSE started

1. If we get a deployment closed event, just bail
2. If we get a lease closed event, just bail
3. Don't return the error for context cancelled if all leases are reporting as OK by providers
4. Run service status after all leases are up and dump the result to standard out.
5. Don't log the same service status multiple times
6. Use a context with a deadline, tie everything to that. Poll "forever" or until the deadline is reached

### CODE REVIEW

File: deploy/cmd/create.go [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+// createCmd represents the create command
+func createCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create [sdl-file]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Create a deployment on the akash network",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			timeoutDuration := viper.GetDuration(FlagTimeout)
+			endAt := time.Now().Add(timeoutDuration)
+			ctx, cancel := context.WithDeadline(context.Background(), endAt)
```
Comment (troian) [inline deploy/cmd/create.go:49]: i think it is better to use `cmd.Context()`

File: deploy/cmd/create.go [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+		if err != nil {
+			cancel()
+			return nil, err
+		}
+
+		servicesChecked := make(map[string]bool)
+
+		return func() error {
+			err = retry.Do(func() error {
+				ls, err := gclient.LeaseStatus(context.Background(), leaseID)
```
Comment (troian) [inline deploy/cmd/create.go:261]: `ctx` from parameters?

File: deploy/cmd/deployment-data.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	}
+	mani, err := sdlSpec.Manifest()
+	if err != nil {
+		return nil, err
+	}
+	ver, err := sdl.ManifestVersion(mani)
+	if err != nil {
+		return nil, err
+	}
+	id, err := dcli.DeploymentIDFromFlags(flags, clientCtx.GetFromAddress().String())
```
Comment (troian) [inline deploy/cmd/deployment-data.go:114]: use `DeploymentIDFromFlagsForOwner` instead. considering `--from` is required to load client certificate `--owner` is not necessary

File: deploy/cmd/create.go (hunk 1 above)
Comment (hydrogen18) [reply deploy/cmd/create.go:49]: Good point, I'll give that a try

**Comment by hydrogen18** [comment]
OK, I think I've addressed most of the comments now

**Review by troian — APPROVED** [review]

**Review by boz — APPROVED** [review]
:rocket:

**Pull request merged**